import_tracker/rest.py

```diff
@@ -76,34 +81,54 @@ def getImports(query=None, user=None, unique=False, limit=None, offset=None, sor
     return processCursor(cursor, user)
 
 
-def moveLeafFiles(folder, user, assetstore):
+def moveLeafFiles(folder, user, assetstore, progress, job):
+    job = Job().load(job['_id'], force=True)
+    if job['status'] == JobStatus.CANCELED:
+        raise ImportTrackerCancelError()
+
     Folder().updateFolder(folder)
 
+    # only move files that are not already in the assetstore
+    unique_clause = {'assetstoreId': {'$ne': ObjectId(assetstore['_id'])}}
+
     folder_item = Item().findOne({
         'folderId': folder['_id'],
     })
-    unique_clause = {'assetstoreId': {'$ne': ObjectId(assetstore['_id'])}}
+    if not folder_item:
+        raise RestException('Folder %s has no item' % folder['_id'])
 
     child_folders = Folder().childFolders(folder, 'folder', user=user)
     child_items = Folder().childItems(folder, filters=unique_clause)
 
-    results = []
-    for attached_file in File().find({
-        'attachedToId': folder_item['_id'],
-        **unique_clause
-    }):
-        setResponseTimeLimit(86400)
-        results.append(Upload().moveFileToAssetstore(attached_file, user, assetstore))
+    # get all files attached to an object
+    def getAttached(attachedToId):
+        uploads = []
+        for attached_file in File().find({'attachedToId': attachedToId, **unique_clause}):
+            setResponseTimeLimit(86400)
+            upload = Upload().moveFileToAssetstore(attached_file, user, assetstore,
+                                                   progress=progress)
+            uploads.append(upload)
+        return uploads
+
+    # upload all files attached to the current folder
+    uploads = getAttached(folder_item['_id'])
 
     for item in child_items:
+        # upload all attached files for each item
+        uploads += getAttached(item['_id'])
+
         setResponseTimeLimit(86400)
         for file in File().find({'itemId': ObjectId(item['_id']), **unique_clause}):
-            results.append(Upload().moveFileToAssetstore(file, user, assetstore))
+            message = f'Moving {folder["name"]}/{file["name"]}\n'
+            job = Job().updateJob(job, log=f'{time.strftime("%Y-%m-%d %H:%M:%S")} - {message}',
+                                  progressMessage=message, status=JobStatus.RUNNING)
```

Comment:
We don't want to send progress through the job mechanism.  rather, we want to send it through the progress context.  Further, the job is already running, so this could be changed to:
```
job = Job().updateJob(job, log=f'{time.strftime("%Y-%m-%d %H:%M:%S")} - {message}')
progress.update(message=message)
```